runestone/unittest_base.py

```diff
@@ -213,6 +215,13 @@ def module_fixture_maker(module_path, return_mf=False, exit_status_success=True)
 
 # Provide a base test case which sets up the `Selenium <http://selenium-python.readthedocs.io/>`_ driver.
 class RunestoneTestCase(unittest.TestCase):
+    # Wait until a Runestone component has finished rendering itself, given the ID of the component.
+    def wait_until_ready(self, id):
```

Comment:
Would there be value in doing this as a "real" event like we do for `runestone:login-complete` ?  This seems like a great selenium hack but would it work outside of selenium?  I could see such an event being potentially very useful for all components.

Reply:
Good point; I like it. How about creating a JS function such as `indicate_component_ready(element, id)` which indicates that the HTML `element` whose ID is `id` is ready? This function can then send an event, update a class, etc. Changing only this function would allow us to modify this behavior (pick a different signal, or store ready components in a dict, etc.).